src/asm/fstack.c

```diff
@@ -115,6 +124,29 @@ static int32_t popcontext(void)
 			sym_UseCurrentMacroArgs();
 			sym_SetMacroArgID(nMacroCount++);
 			sym_UseNewMacroArgs();
+
+			/* Increment REPT count in file path */
+			pREPTIterationWritePtr =
+				strrchr(tzCurrentFileName, '~') + 1;
+			nREPTIterationNo =
+				strtoul(pREPTIterationWritePtr, NULL, 10);
+			nNbCharsLeft = sizeof(tzCurrentFileName)
+				- (pREPTIterationWritePtr - tzCurrentFileName);
+			nNbCharsWritten = snprintf(pREPTIterationWritePtr,
+						   nNbCharsLeft, "%lu",
+						   nREPTIterationNo + 1);
+			if (nNbCharsWritten > nNbCharsLeft) {
```

Comment:
This should be `>=` instead of `>`. The return value doesn't count the null terminator, but the value passed to `snprintf()` (`nNbCharsLeft`) does count it.

Reply:
> The functions snprintf() and vsnprintf() do not write  more  than  size
>        bytes  (including the terminating null byte ('\0')).  If the output was
>        truncated due to this limit, then the return value  is  the  number  of
>        characters  (excluding the terminating null byte) which would have been
>        written to the final string if enough space had been available.   **Thus,
>        a  return  value  of  size or more means that the output was truncated.**

Looks like I misread, oops.